docs/dex-specification.md

```diff
@@ -227,6 +227,8 @@ Each Dex smart contract for the token pair manages a [liquidity pool](#liquidity
 
 The contract holds the following information in the variables: the reserves for both tokens, the timestamp of the latest block, the latest cumulative prices of both tokens, and the value of the `k` [invariant](#invariant) for the pair of tokens calculated as the product of both reserves.
 
+**Warning**: Pair contracts relying on the price changes might not function correctly after the year 2038 (check [Year 2038 Problem](https://en.wikipedia.org/wiki/Year_2038_problem)). This is due to the `_update` function (used to update reserves and price accumulators) casting the `block.timestamp` to `uint32`, which will wrap around in the year 2038. This only affects the `price0CumulativeLast` and `price1CumulativeLast` variables (latest cumulative prices of tokens in the pair), which are used for external price reporting. This is not detrimental to the DEX functionality.
```

Comment:
Not pair contracts but other smart contracts relying on the price functionality. All pair contracts contain this functionality (and the problem itself) but do not fully rely on it (as it is mentioned below, this only affects the price{0,1}CumulativeLast variables , which are used for **external price reporting** and is not detrimental to the functionality of the DEX). Basically, pair contracts are still going to work but contain a bug reporting wrong price{0,1}CumulativeLast prices, other smart contracts using this price reporting functionality would not operate correctly due to the wrong prices they get.